### ISSUE TITLE
run.sh enthält Windows-line-endings

### ISSUE THREAD

**Issue by FelixSFD** [axelvondreden/dms#135]
Die run.sh kann unter Ubuntu 18.04 nicht ausgeführt werden, da sie die falschen Zeilenumbrüche enthält. Dadurch bekommt man die Meldung:

> run.sh: 19: run.sh: Syntax error: end of file unexpected (expecting "then")

Lösung: UNIX-Zeilenumbrüche verwenden

Hier ist ein Befehl beschrieben, der unter Linux-Systemen funktionieren sollte
https://superuser.com/a/52140/703174

**Comment by HeyDennis** [comment]
Willkommen im Team mein Freund :vulcan_salute:
Das wird wohl die Lösung für #130  sein :thumbsup:

**Comment by HeyDennis** [comment]
Habe jetzt mal axl eingetragen... Darfst aber auch gerne du übernehmen.

**Comment by axelvondreden** [comment]
Willkommen @FelixSFD .
Falls du tatsächlich mitwirken willst nehme ich dich gern als contributor auf. Sollten aber vorher vielleicht noch quatschen weil mein Workflow grade eher auf 1 Person ausgelegt ist ;)
Pull-Requests auf den dev branch sollten aber ganz gut funktionieren.

Was ich komisch finde:
Mein Testsystem ist auch eine debian-box und ich hatte noch nie Probleme mit line-endings. Hast du die zip unter windows entpackt oder direkt in linux runtergeladen? (ich meine @HeyDennis hatte da mal einen ähnlichen Fall).

Grundsätzlich kann ich natürlich zur Sicherheit auch vor den releases die Konvertierung nochmal laufen lassen.

Langfristig gefällt mir die aktuelle Lösung eh nicht besonders gut. 
Vielleicht wären separate releases pro Plattform die bessere Lösung, aber fürs Erste lass ich es mal so.

**Comment by HeyDennis** [comment]
>Hast du die zip unter windows entpackt oder direkt in linux runtergeladen? (ich meine @HeyDennis hatte da mal einen ähnlichen Fall).

Genau. Bisher musste ich es immer erst auf der Pi entpacken -> dann hat es funktioniert...
Daher denke ich liegt der aktuelle Fehler von #130 wo anders.

**Comment by FelixSFD** [comment]
Ich hab weder richtig Ahnung von Vaadin, noch von Kotlin. ;-)
Vllt. probier ich mal ein paar Sachen (z.B. GitHub Actions) in einem Fork aus. Brauche also erstmal noch keinen Zugriff.

**Comment by FelixSFD** [comment]
Unter Ubuntu 20.04 bekomm ich übrigens beim Entpacken der `run.sh` schon einen Fehler:

> error: invalid zip file with overlapped components (possible zip bomb)

Könnte mit diesem Problem zusammenhängen.

**Comment by axelvondreden** [comment]
> Ich hab weder richtig Ahnung von Vaadin, noch von Kotlin. ;-)
ja das dachte ich mir ;) habe jetzt auch nicht mit großen Code-Änderungen von dir gerechnet ;)

Ich persönlich halte nicht viel von Github Actions (aktuell alles über TeamCity), aber bin immer froh um Input und Anregungen.

**Comment by FelixSFD** [comment]
ok. TeamCity ist natürlich 100 mal besser :-D Wusste nicht, dass du das hast.

Kann es sein, dass du diese Option aktiviert hast? https://stackoverflow.com/a/44440965/4687348
Oder läuft dein Build Agent eh unter Linux?

**Comment by axelvondreden** [comment]
Ohh guter Tip. Muss ich mal schauen, wenn ich zuhause bin.

**Comment by axelvondreden** [comment]
Trotzdem komisch, dass es bei mir unter linux geht. Ich schieb das per SSH direkt von TC auf den Linux-Server und entpacke per `unzip`.
Der Fehler mit der Zip-Bomb kommt mir irgendwie bekannt vor, muss mal schauen ob ich vielleicht auch einfach ignoriere in dem command.

**Comment by FelixSFD** [comment]
mit `unzip` hab ich es auch gemacht. In Docker (Ubuntu 18.04 Image war das glaub ich) kam der 1. Fehler, in einer VM mit 20.04 kam zusätzlich das mit der ZIP-Bomb.

**Comment by axelvondreden** [comment]
also ich werd das mal umstellen, dann kannst mir im nächsten Release gerne Bescheid geben, ob es besser wird.

**Comment by HeyDennis** [comment]
@FelixSFD 
Geht das mittlerweile bei dir ? Falls du keine Nachricht bekommen hast wegen neuer Version
